packages/astro/src/content/types-generator.ts

```diff
@@ -419,6 +423,29 @@ async function writeContentFiles({
 				for (const entryKey of Object.keys(collection.entries).sort()) {
 					const dataType = collectionConfig?.schema ? `InferEntrySchema<${collectionKey}>` : 'any';
 					dataTypesStr += `${entryKey}: {\n	id: ${entryKey};\n  collection: ${collectionKey};\n  data: ${dataType}\n};\n`;
+					if(settings.config.experimental.contentCollectionJSONSchema) {
+						if (collectionConfig) {
+							// writing inside loop makes sense for the json schema, since we want to have a file per collection
+							const collectionSchemasDir = new URL('./schemas/collections/', contentPaths.cacheDir);
+							if (!fs.existsSync(collectionSchemasDir)) {
+								fs.mkdirSync(collectionSchemasDir, { recursive: true });
+							}
+	
+							if (collectionConfig?.schema)
+								await fs.promises.writeFile(
+									new URL(`./${collectionKey.replace(/"/g, '')}.json`, collectionSchemasDir),
+									JSON.stringify(
+										zodToJsonSchema(collectionConfig?.schema, {
```

Comment:
`writeFile`, `JSON.stringify` and maybe `zodToJsonSchema` could throw errors. We should catch them.

Reply:
`writeFile` is an I/O operation, so you should always expect the impossible. Even permission issues. It's always good to use a `try/catch`. Here, I suppose `JSON.stringify`, here, is safe if it's that `zodToJsonSchema` returns a serializable JSON. If you use `JSON.stringify` when reading from disk, a `try/catch` is advised because the file might have been tempered.